corehq/apps/data_cleaning/models.py

```diff
@@ -585,11 +573,17 @@ def create_for_session(cls, session, prop_id, data_type, match_type, value=None)
             value=value,
         )
 
-    @property
-    def is_editable_property(self):
-        from corehq.apps.data_cleaning.utils.cases import get_case_property_details
-        property_details = get_case_property_details(self.session.domain, self.session.identifier)
-        return property_details.get(self.prop_id, {}).get('is_editable', True)
+    @classmethod
+    def apply_filters_to_query(cls, session, query):
+        xpath_expressions = []
+        for custom_filter in session.filters.all():
+            query = custom_filter.filter_query(query)
+            column_xpath = custom_filter.get_xpath_expression()
+            if column_xpath is not None:
+                xpath_expressions.append(column_xpath)
+        if xpath_expressions:
+            query = query.xpath_query(session.domain, " and ".join(xpath_expressions))
+        return query
```

Comment:
I'm curious why this was moved into the `BulkEditFilter`. I feel like the session is the one that coordinates how all its filters interact with each other, so placing it in `BulkEditFilter` rather than `BulkEditSession` seems to llimit future flexibility.